stablehlo/dialect/StablehloOps.td

```diff
@@ -707,7 +707,8 @@ def StableHLO_Atan2Op : StableHLO_BinaryElementwiseOp<"atan2",
 }
 
 def StableHLO_ComplexOp: StableHLO_BinaryElementwiseOp<"complex", [Pure,
-    SameOperandsElementType, DeclareOpInterfaceMethods<InferTypeOpInterface>]> {
+    SameOperandsElementType, DeclareOpInterfaceMethods<InferTypeOpInterface>],
+    HLO_Fp32Or64Tensor, HLO_ComplexTensor> {
```

Comment:
I think that explicitly providing OperandType and ResultType here is not needed. The code works the same if this portion of the diff is reverted.